File: BankatiProject_Backend/services/TransactionService/src/main/java/com/example/transactionservice/transaction/entity/Subscription.java
package com.example.transactionservice.transaction.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@Document
public class Subscription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private String id;

    private String userId; // L'utilisateur lié à l'abonnement
    private String agentId;

    private BigDecimal price; // Prix de l'abonnement

    private LocalDateTime startDate; // Date de début de l'abonnement

    private LocalDateTime endDate; // Date de fin de l'abonnement

    private boolean active; // Indique si l'abonnement est actif
}

File: BankatiProject_Front-end/src/app/models/Subscription.ts
export interface Subscription {
  id: string; // Identifiant unique de l'abonnement
  userId: string; // L'utilisateur lié à l'abonnement
  agentId: string; // L'agent lié à l'abonnement
  price: number; // Prix de l'abonnement
  startDate: string; // Date de début de l'abonnement (format ISO string)
  endDate: string; // Date de fin de l'abonnement (format ISO string)
  active: boolean; // Indique si l'abonnement est actif
}
